api/apps/api/src/modules/async-jobs-garbage-collector/scenario-async-jobs.garbage-collector.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@nestjs/common';
+import {
+  CalibrationAsyncJob,
+  CostSurfaceAsyncJob,
+  FeaturesWithPuIntersectionAsyncJob,
+  PlanningAreaProtectedCalculationAsyncJob,
+  PlanningUnitsInclusionAsyncJob,
+  ProtectedAreasAsyncJob,
+  RunAsyncJob,
+  ScenarioCloneAsyncJob,
+  ScenarioExportAsyncJob,
+  ScenarioImportAsyncJob,
+  SpecificationAsyncJob,
+} from './async-jobs';
+import { AsyncJobsGarbageCollector } from './async-jobs.garbage-collector';
+
+@Injectable()
+export class ScenarioAsyncJobsGarbageCollector
+  implements AsyncJobsGarbageCollector {
+  constructor(
+    private readonly calibrationAsyncJob: CalibrationAsyncJob,
+    private readonly costSurfaceAsyncJob: CostSurfaceAsyncJob,
+    private readonly featuresWithPuIntersectionAsyncJob: FeaturesWithPuIntersectionAsyncJob,
+    private readonly planningAreaProtectedCalculationAsyncJob: PlanningAreaProtectedCalculationAsyncJob,
+    private readonly planningUnitsInclusionAsyncJob: PlanningUnitsInclusionAsyncJob,
+    private readonly protectedAreasAsyncJob: ProtectedAreasAsyncJob,
+    private readonly runAsyncJob: RunAsyncJob,
+    private readonly scenarioCloneAsyncJob: ScenarioCloneAsyncJob,
+    private readonly scenarioExportAsyncJob: ScenarioExportAsyncJob,
+    private readonly scenarioImportAsyncJob: ScenarioImportAsyncJob,
+    private readonly specificationAsyncJob: SpecificationAsyncJob,
+  ) {}
+  public async sendFailedApiEventsForStuckAsyncJobs(scenarioId: string) {
+    await Promise.all([
```

Comment:
since this is a more or less fixed list, and even if we want to go for a `for await` approach for projects and scenarios, we could still keep this `Promise.all()` here - or maybe err on the side of caution and process each job type sequentially too, since we can kind of afford to take our time as all of this is scheduled to happen behind the scenes in any case, so whenever it finishes it will be ok, and we can instead prioritise trying to avoid overloading the db (especially when the `api_events` table may have grown a lot)